src/client/client-window.ts

```diff
@@ -4,7 +4,12 @@ import type * as d from '../declarations';
 
 export const win = typeof window !== 'undefined' ? window : ({} as Window);
 
-export const doc = win.document || ({ head: {} } as Document);
+export const doc =
+  win.document ||
+  ({
+    head: {},
+    querySelectorAll: () => [] as unknown[],
+  } as unknown as Document);
```

Comment:
I discussed this also in the discord channel: https://discord.com/channels/520266681499779082/1343523701567787029

Any opinion?

Reply:
Uff, sorry for not checking there. It is hard to keep up with all the servers 🙈 Agree on @johnjenkins comment that we should keep the check where we use `document`. Will update my PR.

Reply:
> Uff, sorry for not checking there. It is hard to keep up with all the servers 🙈 Agree on @johnjenkins comment that we should keep the check where we use `document`. Will update my PR.

No worries just wanted to be sure not to have parallel discussions regarding this topic.